Form/Type/TreeSelectType.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\TreeBrowserbundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * A form type to select a node based on a tree browser.
+ *
+ * The widget option provides 2 different ways to display the form type:
+ *
+ * - browser
+ *     Displays only the tree browsers;
+ * - text
+ *     Displays a text field with a button to select using the tree browser
```

Comment:
what is in the text field? editable text field with the path?

maybe the word "compact" should be here somewhere, as that is the central point of this mode